Bookmark-Manager/src/main/java/com/bookmarkmanager/bookmark/BookmarkService.java

```diff
@@ -44,6 +40,40 @@ public Bookmark addBookmark(Bookmark bookmark) {
     return bookmarkRepository.save(bookmark);
   }
 
+  private void precheckBookmark(Bookmark bookmark) {
```

Comment:
```suggestion
  private void validateBookmark(Bookmark bookmark) {
```

Reply:
This validateBookmark should also be used in the updateBook Method.

Reply:
Added validateBookmark for updateBookmark.